### PR TITLE
add logic for toggling single item [#134478877]

### PR DESCRIPTION
**Pull request by anonrose**
https://www.pivotaltracker.com/story/show/134478877

### CODE REVIEW

File: app/components/widget-icons.js [hunk 1]
```diff
@@ -35,6 +35,15 @@ export default Ember.Component.extend(Reloadable, {
         this.reloadIframe();
       }
     },
+    toggleAddingWidget(item){
+      if (item.get("addCount") < 1){
+        item.incrementProperty('addCount');
+        this.selectedWidgets[item.get("id")] = item.get("addCount");
+      }else if(item.get("addCount") >= 1){
+        item.decrementProperty('addCount');
+        this.selectedWidgets[item.get("id")] = item.get("addCount");
```
Comment (jcstringer) [inline app/components/widget-icons.js]: Assuming item.get("addCount") is always an integer can we just DRY this up to 
```
if (item.get("addCount") < 1){
  item.incrementProperty('addCount');
}else{
  item.decrementProperty('addCount');
}
this.selectedWidgets[item.get("id")] = item.get("addCount");
```

**Comment by anonrose** [comment]
@jcstringer Nice, much cleaner. I was worried about it possibly resolving to undefined, but you're correct I can't think of a case where that applies. Is this good to squash and merge?

File: app/templates/components/widget-icons.hbs [hunk 2]
```diff
@@ -35,7 +35,7 @@
     </div>
     <div class="modal-content">
       <div class="collection">
-        {{#each website.gardenWidgets as |item|}}
+        {{#each (sort-by "name:asc" website.gardenWidgets) as |item|}}
```
Comment (jcstringer) [inline app/templates/components/widget-icons.hbs]: Pretty these are already sorted alphabetically at the server level, why do we need to re-sort on the client side?

**Comment by jcstringer** [comment]
🚢

**Pull request merged**